quilt/tools/package.py

```diff
@@ -71,21 +71,25 @@ def __init__(self, store, user, package, path, contents=None):
             os.mkdir(os.path.join(self._path, self.VERSIONS_DIR))
 
         if contents is None:
-            contents = self._load_contents()
+            contents = self._load_contents(pkghash)
 
         self._contents = contents
 
-    def _load_contents(self):
-        instance_hash = None
-        latest_tag = os.path.join(self._path, self.TAGS_DIR, self.LATEST)
-        if not os.path.exists(latest_tag):
-            msg = "Could not find latest tag for package {0}/{1}"
-            raise PackageException(msg.format(self._user, self._package))
-        
-        with open (latest_tag, 'r') as tagfile:
-            instance_hash = tagfile.read()
+    def _load_contents(self, instance_hash=None):
+        if instance_hash is None:
+            latest_tag = os.path.join(self._path, self.TAGS_DIR, self.LATEST)
+            if not os.path.exists(latest_tag):
+                msg = "Could not find latest tag for package {0}/{1}"
+                raise PackageException(msg.format(self._user, self._package))
+
+            with open (latest_tag, 'r') as tagfile:
+                instance_hash = tagfile.read()
 
         contents_path = os.path.join(self._path, self.CONTENTS_DIR, instance_hash)
+        if not os.path.isfile(contents_path):
+            msg = "{h} is not a valid hash for package {u}/{p}"
```

Comment:
isn't the std errmsg syntax something more like:

$ grep 'Invalid hash' *py */*py | grep package
tools/command.py:        raise CommandException("Invalid hash for package {owner}/{pkg}: {hash}".format(

pls also avoid one letter var/arg/fmt names... ;-)